frontend/src/main/scala/bloop/engine/Interpreter.scala

```diff
@@ -280,24 +285,45 @@ object Interpreter {
     }
   }
 
+  private def validOutputPath(output: AbsolutePath, state: State, extension: String): Boolean =
```

Comment:
Let's model this as a method `reasonOfInvalidPath(output: AbsolutePath, state: State, extension: String): Option[String]` and then log the errors at the call-sites instead of side effecting here.
  
Also, can you add curly braces around the expression?